src/Kiota.Builder/Writers/Java/CodeMethodWriter.cs

```diff
@@ -352,7 +352,11 @@ private static void WriteSetterBody(CodeMethod codeElement, LanguageWriter write
         if (parentClass.GetBackingStoreProperty() is CodeProperty backingStore)
             writer.WriteLine($"this.get{backingStore.Name.ToFirstCharacterUpperCase()}().set(\"{codeElement.AccessedProperty?.Name?.ToFirstCharacterLowerCase()}\", value);");
         else
-            writer.WriteLine($"this.{codeElement.AccessedProperty?.NamePrefix}{codeElement.AccessedProperty?.Name?.ToFirstCharacterLowerCase()} = value;");
+        {
+            string value = String.Equals(codeElement.AccessedProperty?.Type?.Name, "PeriodAndDuration", StringComparison.OrdinalIgnoreCase) ? "PeriodAndDuration.of(value.getPeriod(), value.getDuration());" : "value;";
```

Comment:
I'm not sure I understand the rationale here. Why do we need to make a copy instead of affecting the value directly?

Reply:
@baywet, I believe this is because removing the Immutable annotation on the `PeriodAndDuration` type leads to the spotbugs check failing as the type is interpreted as vulnerability as an externally mutable object.

Reply:
but we just removed that annotation?

Reply:
Yep, spotbugs will flag it as a vulnerability. We could simply add that class to the spotbugs exclusion list? However, I feel that this approach is safer.

Reply:
could we have a copy constructor instead? that'd do this (call of with duration and period) so we end up with less complex code?

Reply:
Would this copy constructor be in the PeriodAndDuration class? 
```Java
    public PeriodAndDuration(@Nonnull PeriodAndDuration periodAndDuration) {
        Objects.requireNonNull(periodAndDuration, "parameter periodAndDuration cannot be null");
        PeriodAndDuration periodAndDurationCopy = PeriodAndDuration.of(periodAndDuration.getPeriod(), periodAndDuration.getDuration());
        this.period = periodAndDurationCopy.getPeriod();
        this.duration = periodAndDurationCopy.getDuration();
    }
```
Is this what you mean within the PeriodAndDuration class?

Reply:
We may also stick to the static factories in the class and add a `ofPeriodAndDuration` factory. 

```java 
    /**
     * Creates an instance based on a PeriodAndDuration.
     * @param periodAndDuration  the {@code PeriodAndDuration}, not null
     * @return the combined {@code PeriodAndDuration}, not null
     */
    @Nonnull
    public static PeriodAndDuration ofPeriodAndDuration(@Nonnull PeriodAndDuration periodAndDuration) {
        Objects.requireNonNull(periodAndDuration, "parameter periodAndDuration cannot be null");
        return PeriodAndDuration.of(periodAndDuration.getPeriod(), periodAndDuration.getDuration());
    }
```

Generated code would then be 

``` Java 
	public void setWorkDuration(PeriodAndDuration value) {
		this._workDuration = PeriodAndDuration.ofPeriodAndDuration(value);
	}
```

Reply:
it also has the benefit of being consistent with the other method

Reply:
Sounds good! Will update this.